irohad/ametsuchi/impl/postgres_block_storage_factory.hpp

```diff
@@ -9,25 +9,25 @@
 #include "ametsuchi/block_storage_factory.hpp"
 
 #include "ametsuchi/impl/postgres_block_storage.hpp"
-#include "ametsuchi/impl/soci_utils.hpp"
-#include "backend/protobuf/proto_block_factory.hpp"
 #include "logger/logger_fwd.hpp"
 
 namespace iroha {
   namespace ametsuchi {
     class PostgresBlockStorageFactory : public BlockStorageFactory {
      public:
       PostgresBlockStorageFactory(
-          soci::session &sql,
-          std::shared_ptr<PostgresBlockStorage::BlockTransportFactory>
-              block_factory,
+          std::shared_ptr<PoolWrapper> pool_wrapper,
+          std::shared_ptr<shared_model::proto::ProtoBlockFactory> block_factory,
+          std::function<std::string()> table_name_provider,
+          bool is_temporary,
```

Comment:
If the previous change is done, this factory will only be used for temporary storage, and `is_temporary` flag will not be required.